src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceTest.java

```diff
@@ -74,6 +77,20 @@ void testCreateLinksForGeneralPartnerReturnsSuccess() throws ServiceException {
         assertEquals(expectedUri, sentSubmission.getLinks().get("self"));
     }
 
+    @Test
+    void testExceptionIsThrownWhenFirstAndSecondNationalityIsTheSame() {
```

Comment:
Unable to test for the specific exception spring MethodArgumentNotValidException, this work when running but the test complains because that the MethodParameter constructor argument is null and stops.

To construct a MethodParameter 2 static method are used
- forExecutable
- forParameter
These require an executable, seems heavy for exceptions to be thrown.

forFieldAwareConstructor
This requires a parameter index , which causes problems when the test runs by throwing a 
java.lang.illegalargumentexception: parameter index needs to be between -1 and -1)
This makes no sense.